### PR TITLE
Add navigation to all domains screen

### LINKED ISSUE #22248: 🍎 [Dynamic Cards] Add domain management feature deep linking support
We need to add deep linking support for the domain management feature so that we ship a card for this feature discovery ref: p1702390092810549-slack-C067H36F011

### PR DESCRIPTION
**Pull request by alpavanoglu**
Fixes #22248 

## Description

This PR adds the function that navigates to All Domains screen. Due to the fact that app-site-association file is needed to be updated for the new universal links to work, we'll need to manually call the function to test the functionality.

## Testing Steps
Call `RootViewCoordinator.sharedPresenter.navigateToAllDomains()` from anywhere within the codebase to trigger the navigation to Me -> All Domains.

Question to Reviewer:
Should the navigation open the `Me` Screen first or should it directly open the all domains screen on wherever the app is currently. I opted for the former as I thought this would also highlight the location of it in case they want to access it again.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

This PR Does not impact UI

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/Utility/Universal Links/Routes+Domains.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+struct DomainsManageRoute: Route {
```
Comment (alpavanoglu) [inline WordPress/Classes/Utility/Universal Links/Routes+Domains.swift]: Should this route sit under `Me` routes or should it be separate as is.

**Comment by hassaanelgarem** [comment]
> Due to the fact that app-site-association file is needed to be updated for the new universal links to work, we'll need to manually call the function to test the functionality.

The file is now updated Ref: #22256

**Comment by hassaanelgarem** [comment]
> Due to the fact that app-site-association file is needed to be updated for the new universal links to work, we'll need to manually call the function to test the functionality.

@alpavanoglu The file is now updated Ref: #22256

File: WordPress/Classes/ViewRelated/Me/Me Main/MeViewController.swift [hunk 2]
```diff
@@ -365,6 +365,14 @@ class MeViewController: UITableViewController {
         navigateToTarget(for: RowTitles.accountSettings)
     }
 
+    /// Selects the All Domains row and pushes the All Domains view controller
+    ///
+    public func navigateToAllDomains() {
+    #if JETPACK
```
Comment (alpavanoglu) [inline WordPress/Classes/ViewRelated/Me/Me Main/MeViewController.swift:371]: Adding this to the function itself will cause us to add more `if Jetpack` checks down the chain. I'd rather go it this way but if we prefer to do it properly, I can add those calls. LMK

File: WordPress/Classes/Utility/Universal Links/Routes+Me.swift [hunk 3]
```diff
@@ -7,6 +7,13 @@ struct MeRoute: Route {
     let jetpackPowered: Bool = false
 }
 
+struct MeAllDomainsRoute: Route {
+    let path = "/me/domains"
+    let section: DeepLinkSection? = .me
+    let action: NavigationAction = MeNavigationAction.allDomains
+    let jetpackPowered: Bool = true
+}
```
Comment (salimbraksa) [inline WordPress/Classes/Utility/Universal Links/Routes+Me.swift:15]: @alpavanoglu I think this is not needed, right?

Comment (alpavanoglu) [reply WordPress/Classes/Utility/Universal Links/Routes+Me.swift:15]: Yes I'll remove it

File: WordPress/Classes/Utility/Universal Links/Routes+Domains.swift (hunk 1 above)
Comment (salimbraksa) [reply WordPress/Classes/Utility/Universal Links/Routes+Domains.swift]: I don't have a strong opinion on this, but I think adding the route under `Routes+me.swift` is enough.

```swift
struct MeAllDomainsRoute: Route {
    let path = "/domains/manage"
    let section: DeepLinkSection? = .me
    let action: NavigationAction = MeNavigationAction.allDomains
    let jetpackPowered: Bool = true
}
```

**Review by salimbraksa — COMMENTED** [review]
I've dropped a few comments, but the navigation works as expected. 👍

**Review by salimbraksa — APPROVED** [review]
LGTM! 🚀 

I also tested the case where the user is signed out. In that case, the user is redirected to a web browser. That's why it's important the URL to be valid imo.

**Pull request merged**